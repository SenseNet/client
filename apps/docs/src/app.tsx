import React from 'react'
import { CssBaseline, Typography } from '@material-ui/core'
import { useRepository } from '@sensenet/hooks-react'
import snLogo from './assets/sensenet_logo_transparent.png'

export const App: React.FunctionComponent = () => {
  const repo = useRepository()
  return (
    <div
      style={{
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        width: '100%',
        height: '100%',
        flexDirection: 'column',
        backgroundPosition: 'center',
        backgroundRepeat: 'no-repeat',
        backgroundImage: `url(${snLogo})`,
        backgroundSize: 'auto',
      }}>
      <CssBaseline />
      <Typography variant="h3" gutterBottom>
        Welcome, on {repo.configuration.repositoryUrl} ðŸ˜Ž
      </Typography>
    </div>
  )
}
