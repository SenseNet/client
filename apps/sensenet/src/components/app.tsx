import { CssBaseline } from '@material-ui/core'
import React from 'react'
import AppProviders from './app-providers'
import { Dialogs } from './dialogs'
import { ErrorBoundary } from './error-boundary'
import { DesktopLayout } from './layout/DesktopLayout'
import { MainRouter } from './MainRouter'
import { NotificationComponent } from './NotificationComponent'

export function App() {
  return (
    <ErrorBoundary>
      <AppProviders>
        <CssBaseline />
        <DesktopLayout>
          <MainRouter />
        </DesktopLayout>
        <NotificationComponent />
        <Dialogs />
      </AppProviders>
    </ErrorBoundary>
  )
}
