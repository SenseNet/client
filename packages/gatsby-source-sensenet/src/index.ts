export * from './gatsby-node'
export * from './utils'
