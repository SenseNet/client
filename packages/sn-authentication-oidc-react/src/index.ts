import { UserManager, UserManagerSettings, WebStorageStateStore } from 'oidc-client'
export * from './components/authentication-provider'
export * from './components/oidc-secure'
export * from './use-authentication'
export { WebStorageStateStore, UserManagerSettings, UserManager }
