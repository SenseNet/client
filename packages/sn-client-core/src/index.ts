export * from './Authentication'
export * from './Models'
export * from './Repository'
export * from './Schemas'
