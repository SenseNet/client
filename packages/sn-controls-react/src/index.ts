/**
 * @module sn-controls-react
 * @description General classes, modules and methods for sn-controls-react package
 */

export * from './ReactControlMapper'
export * from './fieldcontrols'
export * from './viewcontrols'
