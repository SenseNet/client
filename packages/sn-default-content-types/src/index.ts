export * from './ActionModel'
export * from './ComplexTypes'
export * from './DefaultContentTypes'
export * from './DefaultSchemas'
export * from './Enums'
export * from './FieldSettings'
export * from './Security'
export * from './type-guards'
