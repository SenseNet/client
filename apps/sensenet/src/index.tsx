import CssBaseline from '@material-ui/core/CssBaseline'
import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'
import { HashRouter, Route } from 'react-router-dom'
import { DesktopLayout } from './components/layout/DesktopLayout'
import { MainRouter } from './components/MainRouter'
import { NotificationComponent } from './components/NotificationComponent'
import {
  ContentRoutingContextProvider,
  InjectorContext,
  LocalizationProvider,
  PersonalSettingsContextProvider,
  RepositoryContextProvider,
  ResponsiveContextProvider,
  SessionContextProvider,
  ThemeProvider,
} from './context'
import { LoggerContextProvider } from './context/LoggerContext'
import { CommandProviderManager } from './services/CommandProviderManager'
import { CheatCommandProvider } from './services/CommandProviders/CheatCommandProvider'
import { CustomActionCommandProvider } from './services/CommandProviders/CustomActionCommandProvider'
import { HelpCommandProvider } from './services/CommandProviders/HelpCommandProvider'
import { HistoryCommandProvider } from './services/CommandProviders/HistoryCommandProvider'
import { InFolderSearchCommandProvider } from './services/CommandProviders/InFolderSearchCommandProvider'
import { NavigationCommandProvider } from './services/CommandProviders/NavigationCommandProvider'
import { QueryCommandProvider } from './services/CommandProviders/QueryCommandProvider'
import { diMiddleware, store } from './store'
import './style.css'
import theme from './theme'
import './utils/errorToJson'
import './utils/InjectorExtensions'

console.log(
  `%c@sensenet app v${process.env.APP_VERSION}
Branch ${process.env.GIT_BRANCH}
Commit '${process.env.GIT_COMMITHASH}' `,
  'color: #16AAA6; border-bottom: 1px solid black',
)

const injector = diMiddleware['injector']

diMiddleware
  .getInjectable(CommandProviderManager)
  .RegisterProviders(
    CheatCommandProvider,
    CustomActionCommandProvider,
    HelpCommandProvider,
    HistoryCommandProvider,
    InFolderSearchCommandProvider,
    NavigationCommandProvider,
    QueryCommandProvider,
  )

ReactDOM.render(
  <CssBaseline>
    <Provider store={store}>
      <InjectorContext.Provider value={injector}>
        <LoggerContextProvider>
          <PersonalSettingsContextProvider>
            <LocalizationProvider>
              <HashRouter>
                <Route path="/:repo?">
                  <RepositoryContextProvider>
                    <ContentRoutingContextProvider>
                      <SessionContextProvider>
                        <ResponsiveContextProvider>
                          <ThemeProvider theme={theme}>
                            <DesktopLayout>
                              <MainRouter />
                              <NotificationComponent />
                            </DesktopLayout>{' '}
                          </ThemeProvider>
                        </ResponsiveContextProvider>
                      </SessionContextProvider>
                    </ContentRoutingContextProvider>
                  </RepositoryContextProvider>
                </Route>
              </HashRouter>
            </LocalizationProvider>
          </PersonalSettingsContextProvider>
        </LoggerContextProvider>
      </InjectorContext.Provider>
    </Provider>
  </CssBaseline>,
  document.getElementById('root'),
)
