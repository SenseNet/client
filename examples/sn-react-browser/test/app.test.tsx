import { shallow } from 'enzyme'
import React from 'react'
import { App } from '../src/app'

describe('The app layout instance', () => {
  it('should be rendered correctly', () => {
    expect(shallow(<App />)).toMatchSnapshot()
  })
})
