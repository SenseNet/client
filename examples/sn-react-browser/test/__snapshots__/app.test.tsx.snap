// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The app layout instance should be rendered correctly 1`] = `
<Fragment>
  <WithStyles(CssBaseline) />
  <UniversalHeader
    appName="sn-react-browser"
    title="Document Browser"
  />
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-1"
    maxWidth="lg"
  >
    <WithStyles(ForwardRef(Button))
      className="makeStyles-button-2"
      href="https://admin.sensenet.com/content/explorer/?path=%2FIT%2FDocument_Library"
      target="_blank"
    >
      Go to connected repository
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-gridContainer-3"
      container={true}
    >
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-gridItem-4"
        item={true}
        xs={12}
      >
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            path="/preview/:documentId"
          />
          <Route
            component={[Function]}
            path="/edit/:documentId/:action?"
          />
        </Switch>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</Fragment>
`;
